src/test/OptionSettlement.t.sol

```diff
@@ -328,16 +336,242 @@ contract OptionSettlementTest is Test, NFTreceiver {
+        vm.startPrank(ALICE);
+        engine.write(optionId, amountToWrite, claimId1);
+        vm.stopPrank();
+
+        // Claim buckets reflect that 30 of 50 written options have been exercised
+        IOptionSettlementEngine.ClaimBucket memory bucket1 =
+            engine.claimBucket(optionId, _getDaysFromBucket(optionWriteTs, 0));
+        assertEq(20, bucket1.amountWritten);
+        assertEq(20, bucket1.amountExercised);
+
+        IOptionSettlementEngine.ClaimBucket memory bucket2 =
+            engine.claimBucket(optionId, _getDaysFromBucket(optionWriteTs, 1));
+        assertEq(20, bucket2.amountWritten);
+        assertEq(10, bucket2.amountExercised);
+
+        IOptionSettlementEngine.ClaimBucket memory bucket3 =
+            engine.claimBucket(optionId, _getDaysFromBucket(optionWriteTs, 2));
+        assertEq(10, bucket3.amountWritten);
+        assertEq(0, bucket3.amountExercised);
+
+        // warp to expiry
+        vm.warp(block.timestamp + 2 * ONE_DAY_SECONDS);
+
+        // redeeming claim2 (written on day 0 for 10) should return 10 * exercise amount
+        // and 0 * underlying amount
+        _redeemAndAssertClaimAmounts(
+            ALICE,
+            claimId1,
+            option,
+            15, // exercised
+            15 // unexercised
+        );
+        _redeemAndAssertClaimAmounts(
+            ALICE,
+            claimId2,
+            option,
+            10, // exercised
+            0 // unexercised
+        );
+        _redeemAndAssertClaimAmounts(
+            ALICE,
+            claimId3,
+            option,
+            5, // exercised
+            5 // unexercised
+        );
+    }
+
+    function _redeemAndAssertClaimAmounts(
+        address claimant,
+        uint256 claimId,
+        IOptionSettlementEngine.Option memory option,
+        uint256 amountExercised,
+        uint256 amountUnexercised
+    ) internal {
+        vm.startPrank(claimant);
+        uint256 exerciseAssetBalance = ERC20(option.exerciseAsset).balanceOf(claimant);
+        uint256 underlyingAssetBalance = ERC20(option.underlyingAsset).balanceOf(claimant);
+
+        engine.redeem(claimId);
+
+        assertEq(
+            ERC20(option.exerciseAsset).balanceOf(claimant),
+            exerciseAssetBalance + amountExercised * option.exerciseAmount
+        );
+
+        assertEq(
+            ERC20(option.underlyingAsset).balanceOf(claimant),
+            underlyingAssetBalance + amountUnexercised * option.underlyingAmount
+        );
+        vm.stopPrank();
+    }
+
     // **********************************************************************
     //                            EVENT TESTS
     // **********************************************************************
 
-    function testEvent_newOptionType() public {
-        vm.expectEmit(false, true, true, true); // ignore 1st topic for now (TODO calculate optionId)
+    function testEventNewOptionType() public {
```

Comment:
Thanks for updating these test names, I will stick to your styling convention going forward 👍

Reply:
if you prefer that naming convention, super open to that too, its just that when I ran build/lint there were some warnings that it was spitting out. might be an artifact of my local though, since the checks must have passed when you submitted the PR to get them in in the first place.

Reply:
Let's keep as you have for the time-being, just to minimize change rn